CONTRIBUTING.md

```diff
@@ -1,2 +1,132 @@
+
+## Code of Conduct
+
+PX Blue has adopted the [Contributor Covenant](https://www.contributor-covenant.org/) as its Code of Conduct, and we expect project participants to adhere to it.
+Please read [the document](https://github.com/pxblue/.github/blob/master/CODE_OF_CONDUCT.md) so that you can understand what will and will not be tolerated.
+
+## Contributions
+
+There are many ways to contribute to PX Blue and code contribution is one aspect of it. Documentation improvements are as important as code changes.
+
+- Reporting and fixing bugs, adding features, and improving our documentation.
+- Add new code examples or new component.
+- Help translate and update API documentation.
+
+## Your first Pull Request
+
+If you decide to fix an issue, please check the comments in the issues section on the repository in case somebody is working on a fix. If nobody is working on it, please leave a comment stating that you have started to work on it so other people don’t accidentally duplicate your effort. If you decide to fix an issue that has not been logged, please log the issue first.
+
+## Branch Naming and Pull Request Workflow
+
+### Branch Naming Conventions:
+
+Prefixed with feature or bug and use dashes as the delimiter.
+Only use lowercase in branch names.
+Be accurate as possible in branch naming.
+
+```
+feature/my-topic-branch or bug/fix-topic-branch
+```
+
+### Commit Message Conventions:
+
+Capitalize the subject line and do not end the subject with a period.
+Use the imperative mood in the subject line. `This commit will <your commit message>`.
+
+### Pull Request:
+
+- Bug branch pull request description will reference the issue# fixed.
+- Feature branch pull request description will explain what the feature is and changes proposed.
+
+## Sending a Pull Request
+
+PX Blue is a community with many projects, so pull requests are always welcome. If you are working on a complete change, please open an issue first to discuss it with the maintainers of PX Blue. Keep pull requests small to have the best chance of getting accepted.
+
+```
+Don't bundle more than one feature or bug fix per pull request.
+```
+
+## Get started
+
+1. Fork the PX Blue repository to your GitHub account.
+
+2. Clone the fork repository to your local machine and add upstream remote:
+
+```
+git clone https://github.com/<your-user-id>/<repository> -b <branch-name>
+cd <repository>
+git remote add upstream https://github.com/pxblue/<repository-name>
+```
+
+3. Synchronize your local branch with the upstream one:
+
+```
+git pull upstream <branch>
+```
+
+4. Install the dependencies:
+
+```
+yarn install
+```
+
+5. Create a new branch:
+
+```
+git checkout -b feature/my-topic-branch
+Or
+git checkout -b bug/my-topic-branch
+```
+
+6. Make changes, commit and push to your fork:
+
+```
+git push –set upstream origin <your branch name>
+```
+
```

Comment:
Do we want to add a step here to tell them to lint/prettier their code and make sure all of the tests pass?

Reply:
They should probably know to do this before they get to the PR stage and have their PR rejected